kyaml/kio/byteio_writer.go

```diff
@@ -6,13 +6,13 @@ package kio
 import (
 	"encoding/json"
 	"io"
-
+	"path/filepath"
 	"sigs.k8s.io/kustomize/kyaml/errors"
 	"sigs.k8s.io/kustomize/kyaml/kio/kioutil"
 	"sigs.k8s.io/kustomize/kyaml/yaml"
 )
 
-// Writer writes ResourceNodes to bytes.
+// ByteWriter writes ResourceNodes to bytes.
```

Comment:
It would be useful to have the JSON vs YAML behavior (as a function of file name annotations) described up here.

Reply:
since you touched the line :)

thanks BTW